tools/ARIAtools/tsSetup.py

```diff
@@ -205,52 +238,46 @@ def generate_stack(aria_prod, input_files, output_file_name,
 
     # Set up single stack file
     if not os.path.exists(os.path.join(workdir, 'stack')):
-        print('Creating directory: {os.path.join(workdir, "stack")}')
+        print('Creating directory: ', os.path.join(workdir, "stack"))
         os.makedirs(os.path.join(workdir, 'stack'))
-    else:
-        print(f'Directory {os.path.join(workdir, "stack")} already exists.')
 
-    if input_files in ['unwrappedPhase', 'unwrapped', 'unw']:
+    # MG; did a little clean up below
+
+    # MG: find pattern in string, with special case for tropo
+    # take into account variation in input name
+    # e.g. [unwrappedPhase, unwrapPhase, unwrap]
+    # NOTE is it necessary to have the option for different names
+    # as they are hardcoreded, maybe because of different nc version, double-check?
+    _find_match = lambda x: re.findall('[A-Z][^A-Z]*', x)[0] \
```

Comment:
This  could be moved to a separate script that check the nc version, input files, find stored layers in the container and  output the layers names for futher processing. Maybe for a next update :D